### DISCUSSION TITLE
Fast traversal / acceleration

### DISCUSSION THREAD

**Discussion by jmsaltzman** [LennartHennigs/ESPRotary#37] [Ideas]
I'm remaking an old project with from-scratch code, and this has worked well for me: thank you!!

I set a callback to rotate() (below) with setChangedHandler() to let me navigate a larger range of values: bigger steps when turning the encoder quickly. It's pretty simple and abuses resetPosition(), but it has proved reliable :)  I tried this first with the setIncrement() function-- setting a larger increment when dialing fast-- but then it would take the bigger step for the next slow click of the dial. It would be cool to have acceleration of some kind in the library, but this does the trick:

```
#define XTIME           75  // ms, spin faster and use a different increment 
#define XFACTOR         5   // encoder increment if spinning fast
...

void rotate(ESPRotary& r) {
  // Multiply if switched fast
  if (millis() < last_dial_time + XTIME) {
    if (r.directionToString(r.getDirection()) == "LEFT") {
      r.resetPosition(r.getPosition() - (XFACTOR - 1), false);
    }
    else {  // RIGHT
      r.resetPosition(r.getPosition() + (XFACTOR - 1), false);
    }
  }
  last_dial_time = millis();
  dial_tenths = r.getPosition();
  }
}
```

I wonder if there's a way to use setIncrement() like this that I couldn't figure out, or if there's a better way to get larger increments with faster turning. Thanks again!

**Comment by LennartHennigs** [comment]
Hey @jmsaltzman, 
I updated the library and it now supports the speedup mode.
Thank you for your suggestion!
I made it (finally) part of a larger rewrite.
Cheers!

**Reply by jmsaltzman** [reply]
Awesome, thank you! Away from home now and looking forward to folding this into my project when I return :)
